nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/test/java/org/apache/nifi/web/util/SnippetUtilsTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.web.util;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import org.apache.nifi.util.ComponentIdGenerator;
+import org.junit.Test;
+
+public class SnippetUtilsTest {
+
+    /*
+     * This test validates condition where component is being replicated across
+     * the cluster
+     */
+    @Test
+    public void validateWithSameSeedSameInceptionIdSameInstanceId() throws Exception {
+        Method generateIdMethod = SnippetUtils.class.getDeclaredMethod("generateId", String.class, String.class,
```

Comment:
Can we just make this 'generateId' method package-level and final, rather than private? I realize that it changes the encapsulation, but using reflection in unit tests is more brittle, as any refactoring breaks the unit tests. This is also the model that we have typically followed in NiFi to-date, rather than using 'private' w/ reflection.